sql-plugin/src/main/spark311/scala/com/nvidia/spark/rapids/shims/Spark31XShims.scala

```diff
@@ -339,7 +339,9 @@ abstract class Spark31XShims extends Spark31Xuntil33XShims with Logging {
           override def tagPlanForGpu(): Unit = GpuFileSourceScanExec.tagSupport(this)
 
           override def convertToCpu(): SparkPlan = {
-            wrapped.copy(partitionFilters = partitionFilters)
+            val cpu = wrapped.copy(partitionFilters = partitionFilters)
+            cpu.copyTagsFrom(wrapped)
+            cpu
```

Comment:
We have the same code for all shims. Can we extract that to a base class/function and get rid of all these repeated ones?

Reply:
Is it OK if that's tracked as a followup?  There's a lot more than this method duplicated between shims, and this is trying to unblock the 24.02 release.